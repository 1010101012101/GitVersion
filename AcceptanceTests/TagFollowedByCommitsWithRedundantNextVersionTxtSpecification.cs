using GitHubFlowVersion.AcceptanceTests.Helpers;
using TestStack.BDDfy;

namespace GitHubFlowVersion.AcceptanceTests
{
    using ApprovalTests;
    using Xunit;

    public class TagFollowedByCommitsWithRedundantNextVersionTxtSpecification : IUseFixture<RepositoryFixture>
    {
        private ExecutionResults _result;
        private const string TaggedVersion = "1.0.3";
        private int _numCommitsToMake;
        RepositoryFixture _data;
        private const string NextVersionTxtVersion = "1.0.0";

        public void GivenARepositoryWithASingleTag()
        {
            _data.Repository.MakeATaggedCommit(TaggedVersion);
        }

        public void AndGivenRepositoryHasAnotherXCommits()
        {
            _data.Repository.MakeCommits(_numCommitsToMake);
        }

        public void AndGivenRepositoryHasARedundantNextVersionTxtFile()
        {
            _data.Repository.AddNextVersionTxtFile(NextVersionTxtVersion);
        }
        
        public void WhenGitHubFlowVersionIsExecuted()
        {
            _result = GitVersionHelper.ExecuteIn(_data.RepositoryPath);
        }

        public void ThenAZeroExitCodeShouldOccur()
        {
            _result.AssertExitedSuccessfully();
        }

        public void AndTheCorrectVersionShouldBeOutput()
        {
            Approvals.Verify(_result.Output, Scrubbers.GuidScrubber);
        }

        [Fact]
        public void WithOneCommit()
        {
            _numCommitsToMake = 1;
            this.BDDfy();
        }

        [Fact]
        public void WithTenCommit()
        {
            _numCommitsToMake = 10;
            this.BDDfy();
        }

        public void SetFixture(RepositoryFixture data)
        {
            _data = data;
        }
    }
}
