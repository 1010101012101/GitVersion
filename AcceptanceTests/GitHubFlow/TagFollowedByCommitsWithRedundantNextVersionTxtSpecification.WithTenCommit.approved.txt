{
  "Major":1,
  "Minor":0,
  "Patch":4,
  "PreReleaseTag":"",
  "PreReleaseTagWithDash":"",
  "BuildMetaData":10,
  "FullBuildMetaData":"10.Branch.master.Sha.000000000000000000000000000000000000000",
  "MajorMinorPatch":"1.0.4",
  "SemVer":"1.0.4",
  "SemVerPadded":"1.0.4",
  "AssemblySemVer":"1.0.4.0",
  "FullSemVer":"1.0.4+10.Branch.master.Sha.000000000000000000000000000000000000000",
  "FullSemVerPadded":"1.0.4+10.Branch.master.Sha.000000000000000000000000000000000000000",
  "ClassicVersion":"1.0.4.10",
  "BranchName":"master",
  "Sha":"000000000000000000000000000000000000000"
}
